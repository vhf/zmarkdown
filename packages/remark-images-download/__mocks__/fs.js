const fs = jest.genMockFromModule('fs')

fs.stat = (path, callback) => {
  const err = null
  const stats = {
    isDirectory () {
      return true
    }
  }
  process.nextTick(() => callback(err, stats))
}

fs.mkdir = (path, callback) => {
  process.nextTick(() => callback())
}

module.exports = fs
