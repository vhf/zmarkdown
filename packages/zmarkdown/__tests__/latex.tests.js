/* eslint-disable no-console */
import {readdirSync as directory, readFileSync as file} from 'fs'
import {join} from 'path'
import dedent from 'dedent'
import remarkConfig from '../remark-config'
import rebberConfig from '../rebber-config'

remarkConfig.noTypography = true
remarkConfig.ping.pingUsername = () => false

const zmarkdown = require('../')

const renderString = (config = {remarkConfig, rebberConfig}) =>
  (input) =>
    zmarkdown(config, 'latex').renderString(input)

const base = join(__dirname, 'fixtures/latex')
const fixtures = directory(base).reduce((tests, contents) => {
  const parts = contents.split('.')
  tests[parts[0]] = file(join(base, contents), 'utf-8')
  return tests
}, {})

test('heading', () => {
  const fixture = fixtures['heading']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('html nodes', () => {
  const {content} = renderString()(dedent`
    # foo
    **something <a> else**
  `)

  expect(content).toMatchSnapshot()
})

test('heading with custom config', () => {
  const fixture = fixtures['heading']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('paragraph', () => {
  const fixture = fixtures['paragraph']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('inline-code', () => {
  const fixture = fixtures['inline-code']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('emoticon', () => {
  const fixture = fixtures['emoticon']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('table', () => {
  const fixture = fixtures['table']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('blockquote', () => {
  const fixture = fixtures['blockquote']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('blockquote with custom config', () => {
  const fixture = fixtures['blockquote']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('figure+caption', () => {
  const fixture = fixtures['figure']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('code', () => {
  const fixture = fixtures['code']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('code+caption', () => {
  const fixture = fixtures['figure-code']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('list', () => {
  const fixture = fixtures['list']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('link', () => {
  const fixture = fixtures['link']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

test('link with special characters', () => {
  const {content} = renderString()(dedent`
    [foo](http://example.com?a=b%c^{}#foo)
  `)

  expect(content).toMatchSnapshot()
})

test('link-prepend', () => {
  const fixture = fixtures['link-prepend']
  const {content} = renderString()(fixture)

  expect(content).toMatchSnapshot()
})

Object.keys(fixtures).filter(Boolean).filter(name => name.startsWith('mix-')).forEach(name => {
  const fixture = fixtures[name]

  test(name, () => {
    const {content} = renderString()(fixture.replace(/·/g, ' '))

    expect(content).toMatchSnapshot()
  })
})

test('footnotes', () => {
  const {content} = renderString()(dedent`
    # mytitle A[^footnoteRef]

    [^footnoteRef]: reference in title

    # mytitle B[^footnoterawhead inner]

    # myti*tle C[^foo inner]*

    a paragraph[^footnoteRawPar inner]
  `)
  expect(content).toMatchSnapshot()
})

test('math', () => {
  const {content} = renderString()(dedent`
    A sentence ($S$) with *italic* and inline math ($C_L$) and $$b$$ another.

    $$
    L = \frac{1}{2} \rho v^2 S C_L
    $$

    hehe
  `)
  expect(content).toMatchSnapshot()
})

test('custom-blocks', () => {
  const fixture = fixtures['blocks']
  const {content} = renderString()(fixture.replace(/·/g, ' '))

  expect(content).toMatchSnapshot()
})

test('regression: code block without language', () => {
  const {content} = renderString()(dedent`
    \`\`\`
    a
    \`\`\`
  `)

  expect(content).toMatchSnapshot()
})
