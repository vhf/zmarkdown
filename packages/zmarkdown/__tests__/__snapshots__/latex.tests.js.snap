// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blockquote 1`] = `
"\\\\begin{Quotation}
a quote

a
multiline
quote
\\\\end{Quotation}



\\\\horizontalLine



\\\\begin{Quotation}
a quote
\\\\end{Quotation}



\\\\horizontalLine



\\\\begin{Quotation}
a
multiline
quote
\\\\end{Quotation}

"
`;

exports[`blockquote with custom config 1`] = `
"\\\\begin{Quotation}
a quote

a
multiline
quote
\\\\end{Quotation}



\\\\horizontalLine



\\\\begin{Quotation}
a quote
\\\\end{Quotation}



\\\\horizontalLine



\\\\begin{Quotation}
a
multiline
quote
\\\\end{Quotation}

"
`;

exports[`code 1`] = `
"\\\\begin{CodeBlock}{python}
print('bla')
\\\\end{CodeBlock}



\\\\begin{CodeBlock}[][1,2]{python}
print('bla')
print('bla')
print('bla')
\\\\end{CodeBlock}



\\\\begin{CodeBlock}{text}
a code without lang
\\\\end{CodeBlock}

"
`;

exports[`code+caption 1`] = `
"\\\\begin{codeBlock}[With Source]{python}
print('bla')
\\\\end{codeBlock}



\\\\begin{codeBlock}[With Source][1,2]{python}
print('bla')
print('bla')
print('bla')
\\\\end{codeBlock}

"
`;

exports[`custom-blocks 1`] = `
"\\\\begin{Information}
info
\\\\end{Information}


\\\\begin{Warning}
warning
\\\\end{Warning}


\\\\begin{Spoiler}
secret
\\\\end{Spoiler}


\\\\begin{Question}
question \\\\texttt{coded}
\\\\end{Question}


\\\\begin{Error}
\\\\textbf{error}
foo bar \\\\\\\\
baz
\\\\end{Error}
"
`;

exports[`emoticon 1`] = `
"foo  bar

"
`;

exports[`figure+caption 1`] = `
"\\\\begin{Quotation}[With Source]
a
multiline
quote
\\\\end{Quotation}

"
`;

exports[`footnotes 1`] = `
"\\\\levelTwoTitle{mytitle A\\\\textsuperscript{\\\\ref{footnote:1}}}


\\\\footnotetext[1]{\\\\label{footnote:1} reference in title}


\\\\levelTwoTitle{mytitle B\\\\footnote[2]{\\\\label{footnote:2} footnoterawhead inner}
\\\\protect}


\\\\levelTwoTitle{myti\\\\textit{tle C\\\\footnote[3]{\\\\label{footnote:3} foo inner}
\\\\protect}}


a paragraph\\\\footnote[4]{\\\\label{footnote:4} footnoteRawPar inner}

"
`;

exports[`heading 1`] = `
"\\\\levelTwoTitle{first level title}


\\\\levelThreeTitle{second level title}


\\\\levelFourTitle{third level title}


\\\\levelFiveTitle{fourth level title}


\\\\levelSixTitle{fifth level title}


\\\\levelSevenTitle{sixth level title}


\\\\levelTwoTitle{a \\\\textbackslash{} b}


\\\\levelTwoTitle{a \\\\} b}


\\\\levelTwoTitle{a \\\\} \\\\textbackslash{} b}


\\\\levelTwoTitle{a \\\\} b}
"
`;

exports[`heading with custom config 1`] = `
"\\\\levelTwoTitle{first level title}


\\\\levelThreeTitle{second level title}


\\\\levelFourTitle{third level title}


\\\\levelFiveTitle{fourth level title}


\\\\levelSixTitle{fifth level title}


\\\\levelSevenTitle{sixth level title}


\\\\levelTwoTitle{a \\\\textbackslash{} b}


\\\\levelTwoTitle{a \\\\} b}


\\\\levelTwoTitle{a \\\\} \\\\textbackslash{} b}


\\\\levelTwoTitle{a \\\\} b}
"
`;

exports[`html nodes 1`] = `
"\\\\levelTwoTitle{foo}


\\\\textbf{something <a> else}

"
`;

exports[`inline-code 1`] = `
"a paragraph \\\\texttt{with inline code}.



\\\\texttt{a
multiline
inlinecode}



\\\\texttt{a code with \\\\textbackslash{}}



\\\\texttt{a \\\\textbackslash{}text in \\\\textbackslash{}LaTeX \\\\textbackslash{}\\\\textbackslash{}}

"
`;

exports[`link 1`] = `
"\\\\externalLink{an external link}{https://wikipedia.com}



\\\\externalLink{an internal link}{http://zestedesavoir.com/forums}



\\\\externalLink{an \\\\texttt{external} link}{https://wikipedia.com}

"
`;

exports[`link with special characters 1`] = `
"\\\\externalLink{foo}{http://example.com?a=b\\\\%c\\\\textasciicircum{}\\\\{\\\\}\\\\#foo}

"
`;

exports[`link-prepend 1`] = `
"\\\\externalLink{an external link}{https://wikipedia.com}



\\\\externalLink{an internal link}{http://zestedesavoir.com/forums}



\\\\externalLink{an \\\\texttt{external} link}{https://wikipedia.com}

"
`;

exports[`list 1`] = `
"\\\\begin{itemize}
\\\\item an
\\\\item \\\\textbf{unordered}
\\\\item list
\\\\end{itemize}


\\\\begin{enumerate}
\\\\item an
\\\\item \\\\texttt{ordered}
\\\\item list
\\\\end{enumerate}
"
`;

exports[`math 1`] = `
"A sentence ($S$) with \\\\textit{italic} and inline math ($C_L$) and $$b$$ another.



\\\\[ L = \\\\frac{1}{2} \\\\rho v^2 S C_L \\\\]



hehe

"
`;

exports[`mix-1 1`] = `
"(for convenience,   are replaced with simple single spaces in the tests)



\\\\levelTwoTitle{first 1}


foo \\\\\\\\
bar



Disrupt \\\\sout{asymmetrical} unicorn, pok pok kale chips umami selfies gastropub food truck flannel. Blue \\\\textbf{bottle craft} beer hexagon artisan. Chia gochujang crucifix, \\\\textsuperscript{ready} \\\\textsuperscript{made} hammock \\\\textit{blog succulents} sriracha raw denim scenester cray typewriter fashion axe \\\\textsubscript{art} \\\\textit{party}. Schlitz tacos tilde intelligentsia, unicorn fixie adaptogen beard 8-bit street \\\\textsubscript{art} typewriter.



\\\\levelThreeTitle{\\\\textit{second} 1}


\\\\textbf{Literally selfies tbh lo-fi. Actually health goth ennui, brooklyn retro polaroid sriracha. Kogi live-edge mixtape marfa street \\\\textsubscript{art} synth. Godard synth truffaut selfies, vape fanny \\\\sout{pack} subway tile. Stumptown af pabst, try-hard fam ethical actually four dollar toast. Microdosing kogi brooklyn, locavore jianbing etsy sartorial \\\\textit{YOLO}. Williamsburg salvia photo booth \\\\textsuperscript{readymade} listicle man braid.}



\\\\horizontalLine



\\\\begin{Quotation}[Guru]
Marfa pickled helvetica put a bird on it hot chicken williamsburg.
Edison bulb asymmetrical \\\\texttt{keffiyeh} schlitz iceland put a bird on it hoodie affogato.

Tofu tote bag distillery viral knausgaard, health goth asymmetrical.
\\\\end{Quotation}



Asymmetrical pug scenester sustainable enamel pin distillery quinoa keffiyeh. Flannel humblebrag PBR\\\\&B polaroid banh mi wolf. Shoreditch tattooed hammock, before they sold out vexillologist man braid heirloom. Activated charcoal craft beer cloud bread meh biodiesel pabst.



\\\\levelFourTitle{\\\\sout{third} 1}


Art party keytar godard iceland neutra cronut. Austin \\\\textsuperscript{readymade} semiotics, edison bulb cloud bread adaptogen blue bottle jean shorts DIY. Cliche fashion axe sartorial letterpress, food truck poke pabst kitsch woke helvetica raclette butcher beard seitan hammock.



\\\\texttt{foo
bar}



\\\\begin{Quotation}
\\\\texttt{fo\\\\textbackslash{}o
bar}
\\\\end{Quotation}



Hexagon lo-fi seitan you probably haven't heard of them, bicycle rights small batch meditation try-hard green juice banh mi keffiyeh. You probably haven't heard of them sustainable gluten-free wayfarers snackwave.



\\\\levelFourTitle{third 2}


\\\\levelFiveTitle{\\\\sout{fo}u\\\\textbf{r}t\\\\textit{h} \\\\textit{1}}


Mumblecore kombucha offal, health goth next level before they sold out gluten-free chicharrones keffiyeh. Mumblecore kickstarter hoodie fixie keffiyeh. Microdosing lo-fi taiyaki irony pok pok. Banjo brooklyn umami succulents flexitarian. Swag sartorial scenester synth viral.



\\\\levelSixTitle{fifth 1}


Banjo bespoke subway tile, street \\\\textsubscript{a\\\\textit{r}}\\\\textsuperscript{t} drinking vinegar offal franzen. Pour-over green juice vaporware kale chips. Meggings cronut affogato PBR\\\\&B, \\\\textsubscript{art} party unicorn dreamcatcher schlitz yuccie mixtape 90's thundercats air plant. Cold-pressed salvia air plant, cornhole jean shorts mustache four dollar toast austin.



\\\\levelFourTitle{third 3}


Meditation heirloom chicharrones, sartorial man braid hot chicken sustainable. Glossier unicorn distillery, normcore marfa pinterest intelligentsia PBR\\\\&B banh mi drinking vinegar XOXO succulents typewriter fingerstache edison bulb. Meditation ethical cronut glossier cliche kickstarter.



\\\\levelFiveTitle{fourth 2}


Venmo bushwick food truck chartreuse fam. Everyday carry gastropub glossier, cold-pressed salvia migas keytar. Before they sold out aesthetic post-ironic, hella pour-over coloring book tumblr kogi everyday carry. Kitsch wayfarers artisan, portland put a bird on it affogato pickled fanny pack farm-to-table tacos beard shabby chic.



\\\\levelFiveTitle{fourth 3}


Chambray intelligentsia vape listicle. Pok pok snackwave cronut retro hot chicken, trust fund master cleanse keytar forage dreamcatcher. Taiyaki actually deep v, godard small batch irony lumbersexual unicorn cardigan af. Irony lumbersexual salvia, pitchfork fam snackwave man bun. Kitsch jianbing intelligentsia freegan, waistcoat raw denim cloud bread vice cray etsy listicle skateboard jean shorts.



\\\\levelSixTitle{fifth 2}


Asymmetrical normcore portland, vaporware viral tote bag DIY slow-carb kogi fanny pack. Keffiyeh ennui church-key, irony VHS man bun edison bulb listicle cloud bread try-hard cred lumbersexual lo-fi glossier.



\\\\levelTwoTitle{first 2}


\\\\levelThreeTitle{second 2}


\\\\levelSixTitle{fifth 3}


Messenger bag locavore swag raclette brunch whatever, portland food truck. PBR\\\\&B cred mlkshk, poke letterpress waistcoat celiac. La croix letterpress forage keffiyeh.

"
`;

exports[`mix-2 1`] = `
"(for convenience,   are replaced with simple single spaces in the tests)



\\\\levelTwoTitle{first 1}


\\\\begin{Quotation}[Quotation Source]
Code inside quote

\\\\begin{codeBlock}[code source]{python}
print('bla')
\\\\end{codeBlock}
\\\\end{Quotation}



\\\\begin{codeBlock}[First \\\\keys{CTRL}+\\\\keys{S}]{python}
print('bla')
\\\\end{codeBlock}



Code: Second



\\\\begin{CodeBlock}{python}
print('code without caption')
\\\\end{CodeBlock}

"
`;

exports[`mix-3 1`] = `
"\\\\levelTwoTitle{right-aligned list}



\\\\begin{flushright}
\\\\begin{itemize}
\\\\item item
\\\\item item
\\\\end{itemize}

\\\\end{flushright}


\\\\levelTwoTitle{centered list}



\\\\begin{center}
\\\\begin{itemize}
\\\\item item
\\\\item item
\\\\item item
\\\\end{itemize}

\\\\end{center}


\\\\levelTwoTitle{left list}




\\\\begin{itemize}
\\\\item item
\\\\item item
\\\\item item
\\\\end{itemize}


"
`;

exports[`mix-4 1`] = `
"\\\\begin{longtabu}{|p{\\\\linewidth / 3}|p{\\\\linewidth / 3}|p{\\\\linewidth / 3}|} \\\\hline
Sub & Headings & \\\\abbr{ABBR}{abbreviation} \\\\\\\\ \\\\hline
\\\\multirow{2}{*}{cell \\\\par spans \\\\par rows} & \\\\multicolumn{2}{|c|}{column spanning} \\\\\\\\ \\\\hline
normal & cell \\\\\\\\ \\\\hline
multi \\\\par line \\\\par  \\\\par cells \\\\par too & \\\\multicolumn{2}{|c|}{cells can be \\\\par \\\\textit{formatted} \\\\par \\\\textbf{paragraphs}} \\\\\\\\ \\\\hline

\\\\tableCaption{The new table \\\\abbr{ABBR}{abbreviation} \\\\textsuperscript{\\\\ref{footnote:1}} with \\\\keys{CTRL} + \\\\keys{S}}
\\\\end{longtabu}



\\\\footnotetext[1]{\\\\label{footnote:1} a foot}


\\\\begin{longtabu}{|p{\\\\linewidth / 2}|p{\\\\linewidth / 2}|} \\\\hline
title & image \\\\\\\\ \\\\hline
space & \\\\image{https://i.ytimg.com/vi/lt0WQ8JzLz4/maxresdefault.jpg} \\\\\\\\ \\\\hline
\\\\end{longtabu}



\\\\begin{longtabu}{|p{\\\\linewidth / 2}|p{\\\\linewidth / 2}|} \\\\hline
title & code \\\\\\\\ \\\\hline
inline & \\\\texttt{inline} br \\\\par \\\\texttt{inline} \\\\\\\\ \\\\hline
block & \\\\ref{appendix-1} \\\\\\\\ \\\\hline
\\\\end{longtabu}



\\\\levelTwoTitle{Annexes}


\\\\footnote{\\\\label{appendix-1}}
\\\\begin{CodeBlock}{python}
print('bla')
\\\\end{CodeBlock}

"
`;

exports[`mix-5 1`] = `
"\\\\image{http://www.numerama.com/content/uploads/2016/07/espace.jpg}[espace\\\\textsuperscript{\\\\ref{footnote:1}}]



\\\\footnotetext[1]{\\\\label{footnote:1} Two things are infinite: the universe and human stupidity.}
"
`;

exports[`mix-6 1`] = `
"\\\\image{http://img.youtube.com/vi/8TQIvdFl4aU/0.jpg}[a caption\\\\externalLink{https://www.youtube.com/embed/8TQIvdFl4aU}{https://www.youtube.com/embed/8TQIvdFl4aU}]



\\\\image{http://img.youtube.com/vi/8TQIvdFl4aU/0.jpg}[\\\\externalLink{https://www.youtube.com/embed/8TQIvdFl4aU}{https://www.youtube.com/embed/8TQIvdFl4aU}]



no caption

"
`;

exports[`paragraph 1`] = `
"\\\\levelTwoTitle{first 1}


Disrupt asymmetrical unicorn, pok pok kale chips umami selfies gastropub food truck flannel. Blue bottle craft beer hexagon artisan. Chia gochujang crucifix, readymade hammock blog succulents sriracha raw denim scenester cray typewriter fashion axe art party. Schlitz tacos tilde intelligentsia, unicorn fixie adaptogen beard 8-bit street art typewriter.



\\\\levelThreeTitle{second 1}


Literally selfies tbh lo-fi. Actually health goth ennui, brooklyn retro polaroid sriracha. Kogi live-edge mixtape marfa street art synth. Godard synth truffaut selfies, vape fanny pack subway tile. Stumptown af pabst, try-hard fam ethical actually four dollar toast. Microdosing kogi brooklyn, locavore jianbing etsy sartorial YOLO. Williamsburg salvia photo booth readymade listicle man braid.



Marfa pickled helvetica put a bird on it hot chicken williamsburg. Edison bulb asymmetrical keffiyeh schlitz iceland put a bird on it hoodie affogato. Tofu tote bag distillery viral knausgaard, health goth asymmetrical.



Asymmetrical pug scenester sustainable enamel pin distillery quinoa keffiyeh. Flannel humblebrag PBR\\\\&B polaroid banh mi wolf. Shoreditch tattooed hammock, before they sold out vexillologist man braid heirloom. Activated charcoal craft beer cloud bread meh biodiesel pabst.



\\\\levelFourTitle{third 1}


Art party keytar godard iceland neutra cronut. Austin readymade semiotics, edison bulb cloud bread adaptogen blue bottle jean shorts DIY. Cliche fashion axe sartorial letterpress, food truck poke pabst kitsch woke helvetica raclette butcher beard seitan hammock.



Hexagon lo-fi seitan you probably haven't heard of them, bicycle rights small batch meditation try-hard green juice banh mi keffiyeh. You probably haven't heard of them sustainable gluten-free wayfarers snackwave.



\\\\levelFourTitle{third 2}


\\\\levelFiveTitle{fourth 1}


Mumblecore kombucha offal, health goth next level before they sold out gluten-free chicharrones keffiyeh. Mumblecore kickstarter hoodie fixie keffiyeh. Microdosing lo-fi taiyaki irony pok pok. Banjo brooklyn umami succulents flexitarian. Swag sartorial scenester synth viral.



\\\\levelSixTitle{fifth 1}


Banjo bespoke subway tile, street art drinking vinegar offal franzen. Pour-over green juice vaporware kale chips. Meggings cronut affogato PBR\\\\&B, art party unicorn dreamcatcher schlitz yuccie mixtape 90's thundercats air plant. Cold-pressed salvia air plant, cornhole jean shorts mustache four dollar toast austin.



\\\\levelFourTitle{third 3}


Meditation heirloom chicharrones, sartorial man braid hot chicken sustainable. Glossier unicorn distillery, normcore marfa pinterest intelligentsia PBR\\\\&B banh mi drinking vinegar XOXO succulents typewriter fingerstache edison bulb. Meditation ethical cronut glossier cliche kickstarter.



\\\\levelFiveTitle{fourth 2}


Venmo bushwick food truck chartreuse fam. Everyday carry gastropub glossier, cold-pressed salvia migas keytar. Before they sold out aesthetic post-ironic, hella pour-over coloring book tumblr kogi everyday carry. Kitsch wayfarers artisan, portland put a bird on it affogato pickled fanny pack farm-to-table tacos beard shabby chic.



\\\\levelFiveTitle{fourth 3}


Chambray intelligentsia vape listicle. Pok pok snackwave cronut retro hot chicken, trust fund master cleanse keytar forage dreamcatcher. Taiyaki actually deep v, godard small batch irony lumbersexual unicorn cardigan af. Irony lumbersexual salvia, pitchfork fam snackwave man bun. Kitsch jianbing intelligentsia freegan, waistcoat raw denim cloud bread vice cray etsy listicle skateboard jean shorts.



\\\\levelSixTitle{fifth 2}


Asymmetrical normcore portland, vaporware viral tote bag DIY slow-carb kogi fanny pack. Keffiyeh ennui church-key, irony VHS man bun edison bulb listicle cloud bread try-hard cred lumbersexual lo-fi glossier.



\\\\levelTwoTitle{first 2}


\\\\levelThreeTitle{second 2}


\\\\levelSixTitle{fifth 3}


Messenger bag locavore swag raclette brunch whatever, portland food truck. PBR\\\\&B cred mlkshk, poke letterpress waistcoat celiac. La croix letterpress forage keffiyeh.

"
`;

exports[`regression: code block without language 1`] = `
"\\\\begin{CodeBlock}{text}
a
\\\\end{CodeBlock}

"
`;

exports[`table 1`] = `
"\\\\begin{longtabu}{|c|c|} \\\\hline
1 & 2 \\\\\\\\ \\\\hline
1 & 2 \\\\\\\\ \\\\hline
1 & 2 \\\\\\\\ \\\\hline
1 & 2 \\\\\\\\ \\\\hline
\\\\end{longtabu}



\\\\begin{longtabu}{|c|c|c|} \\\\hline
1 & 2 & 3 \\\\\\\\ \\\\hline
1 & 2 \\\\\\\\ \\\\hline
1 & 2  & 3 \\\\\\\\ \\\\hline
1 & 2  & 3 \\\\\\\\ \\\\hline
\\\\end{longtabu}

"
`;
